app/containers/GamesPage/index.js

```diff
@@ -110,28 +111,24 @@ export class GamesPage extends React.PureComponent { // eslint-disable-line reac
             }, 1000);
         }
 
-        // if (gameAccessToken !== gameAccessToken) {
-        //     // setState gameAccessToken
-        // }
+        if (dataChecking(nextProps, 'gamesPage', 'gameToken', 'success') !== dataChecking(this.props, 'gamesPage', 'gameToken', 'success') && nextProps.gamesPage.gameToken.success) {
+            this.setState({ gameAccessToken: nextProps.gamesPage.gameToken.data.token });
+        }
 
+        if (dataChecking(nextProps, 'gamesPage', 'result') !== dataChecking(this.props, 'gamesPage', 'result') && nextProps.gamesPage.result.success) {
+            this.setState({ gameResultImagelink: nextProps.gamesPage.result.data });
+        }
         // if (availableChance !== availableChance) {
         //     // setState availableChance
         // }
     }
 
-    componentWillReceiveProps = (nextProps) => {
-        if (dataChecking(nextProps, 'gamesPage', 'login', 'success') !== dataChecking(this.props, 'gamesPage', 'login', 'success') && nextProps.gamesPage.login.success) {
-            setTimeout(() => {
-                this.setState({ hideLoginModal: true });
-            }, 1000);
-        }
-    }
-
     onGameComplete = (result) => {
-        alert(JSON.stringify(result));
-
-        // dispatch action(this.state.gameAccessToken, 'success' || 'lose'), PUI https://api.hermo.my/xmas/game, they will return imagelink, show the imagelink
-        // set reducer gameResultImagelink
+        const payload = {
+            score: result,
+            token: this.state.gameAccessToken,
+        };
```

Comment:
wrap the payload inside perfectPairGame, because we might have diff payload for each game, so let's wrap everything in the game itself